src/main/java/org/apache/sling/resourceresolver/impl/mapping/MapEntries.java

```diff
@@ -1027,12 +1044,28 @@ private boolean addEntry(final Map<String, List<MapEntry>> entryMap, final Strin
      */
     private Map<String, Map<String, String>> loadAliases(final ResourceResolver resolver) {
         final Map<String, Map<String, String>> map = new ConcurrentHashMap<>();
-        final String queryString = "SELECT sling:alias FROM nt:base WHERE sling:alias IS NOT NULL";
-        final Iterator<Resource> i = resolver.findResources(queryString, "sql");
-        while (i.hasNext()) {
-            final Resource resource = i.next();
-            loadAlias(resource, map);
-        }
+        String queryString = "SELECT sling:alias FROM nt:base WHERE sling:alias IS NOT NULL";
+		if (this.factory.isForceNoAliasTraversal()) {
+			queryString += " option(traversal fail)";
+		}
+		while (true){
+	        try {
+		        final Iterator<Resource> i = resolver.findResources(queryString, "sql");
+		        while (i.hasNext()) {
+		            final Resource resource = i.next();
+		            loadAlias(resource, map);
+		        }
+		        break;
+	        } catch (Exception e) {
+	        	log.debug("Expected index not available and no traversal enforced", e);
+	            try {
+	                TimeUnit.MILLISECONDS.sleep(getTraversalTimer());
+	            } catch (InterruptedException ex) {
+	                log.warn("Interrupted while sleeping");
```

Comment:
The root exception should also be logged. But this is executed in the thread started in MapEntries.init(), right? This means that a failure will not be propagated to the caller.